core/src/main/java/com/linecorp/armeria/client/proxy/ProxyConfig.java

```diff
@@ -16,83 +16,86 @@
 
 package com.linecorp.armeria.client.proxy;
 
-import static com.linecorp.armeria.client.proxy.NoopProxyConfig.NOOP_PROXY_CONFIG;
+import static com.linecorp.armeria.client.proxy.DisabledProxyConfig.DISABLED_PROXY_CONFIG;
 import static java.util.Objects.requireNonNull;
 
 import java.net.InetSocketAddress;
 
 import javax.annotation.Nullable;
 
-import com.google.common.base.MoreObjects;
-
 import com.linecorp.armeria.client.ClientFactory;
 
 /**
  * Contains base configuration for proxy related settings used in {@link ClientFactory}.
  */
 public abstract class ProxyConfig {
 
+    ProxyConfig() {
+    }
+
     /**
      * Creates a {@code ProxyConfig} configuration for SOCKS4 protocol.
      * @param proxyAddress The proxy address.
      */
-    public static Socks4ProxyConfigBuilder socks4(InetSocketAddress proxyAddress) {
-        return new Socks4ProxyConfigBuilder(requireNonNull(proxyAddress));
+    public static Socks4ProxyConfig socks4(InetSocketAddress proxyAddress) {
+        return new Socks4ProxyConfig(requireNonNull(proxyAddress, "proxyAddress"), null);
     }
 
     /**
-     * Creates a {@code ProxyConfig} configuration for SOCKS5 protocol.
+     * Creates a {@code ProxyConfig} configuration for SOCKS4 protocol.
      * @param proxyAddress The proxy address.
+     * @param username The user name.
      */
-    public static Socks5ProxyConfigBuilder socks5(InetSocketAddress proxyAddress) {
-        return new Socks5ProxyConfigBuilder(requireNonNull(proxyAddress));
+    public static Socks4ProxyConfig socks4(InetSocketAddress proxyAddress, @Nullable String username) {
```

Comment:
Perhaps better making `username` non-null, given that we have the version that does not require `username`?